packages/code-studio/src/plugins/PluginUtils.tsx

```diff
@@ -58,21 +58,21 @@ class PluginUtils {
   static async loadJson(
     jsonUrl: string
   ): Promise<{ plugins: { name: string; main: string }[] }> {
-    return new Promise((resolve, reject) => {
-      const request = new XMLHttpRequest();
-      request.addEventListener('load', () => {
-        try {
-          const json = JSON.parse(request.responseText);
-          resolve(json);
-        } catch (err) {
-          reject(err);
+    return fetch(jsonUrl).then(res => {
+      if (!res.ok) {
+        if (res.status !== 404) {
+          // The browser already logs an error for 404
+          log.error(`Error loading plugins from ${jsonUrl}:`, res.statusText);
         }
-      });
-      request.addEventListener('error', err => {
-        reject(err);
-      });
-      request.open('GET', jsonUrl);
-      request.send();
+        return { plugins: [] };
+      }
+
+      try {
+        return res.json();
```

Comment:
`res.json()` returns a promise, so the `try`/`catch` block here does nothing. Need to `await` it, and should log the error itself as well... but actually, we shouldn't return `{ plugins: [] }` in this case anyway, we should just `throw`. `loadPlugins()` already needs to handle a promise rejection, and this is a good case to just reject the promise. I could also argue not being able to load the manifest is still an error and just throw that, just returning `{ plugins: [] }` if the manifest doesn't exist (possibly with logging a warning.

Reply:
Good catch

As for the other part, I think I initially did it so there was only 1 error in the log if the manifest 404ed. I can throw and it will at least be a better 2nd error than it was before w/ the parsing error message